### PR TITLE
Migrate "components/suggestion/menu_action_provider.jsx" to Typescript

### PR DESCRIPTION
**Pull request by babinderrathi**
#### Summary
Migration of components/suggestion/menu_action_provider.jsx to Typescript

#### Ticket Link

Fixes: https://mattermost.atlassian.net/browse/MM-47070
Fixes: https://github.com/mattermost/mattermost-server/issues/21129

#### Related Pull Requests
NONE

#### Screenshots
NONE

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by sinansonmez — APPROVED** [review]
Good work, thanks for the support. LGTM 🎉

File: components/suggestion/menu_action_provider.tsx [hunk 1]
```diff
@@ -29,12 +31,13 @@ class MenuActionSuggestion extends Suggestion {
 }
 
 export default class MenuActionProvider extends Provider {
-    constructor(options) {
+    options: Array<Record<string, any>>;
```
Comment (M-ZubairAhmed) [inline components/suggestion/menu_action_provider.tsx:34]:
```suggestion
    private options: Array<Record<string, any>>;
```

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
Just one small nit

**Review by M-ZubairAhmed — APPROVED** [review]
💯 It looks good to me. Thank you 🎉

**Pull request merged**